version: "3.9"

services:
  postgres:
    image: postgres:15.2
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - "5432:5432"
    restart: "no"
    volumes:
      - ../src/main/resources/db:/docker-entrypoint-initdb.d:ro
    env_file:
      - .env
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready" ]
      interval: 10s
      timeout: 30s
      retries: 5
  jvm_app:
    build:
      context: ../
      dockerfile: ./env/jvm_app/DockerfileJvmApp
    command: bin/app
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_URL: ${POSTGRES_URL}
      POSTGRES_DB: ${POSTGRES_DB}
      MAIL_HOST: ${MAIL_HOST}
      MAIL_PORT: ${MAIL_PORT}
      MAIL_USERNAME: ${MAIL_USERNAME}
      MAIL_PASSWORD: ${MAIL_PASSWORD}
      COVID_API_KEY: ${COVID_API_KEY}
    ports:
      - "8080:8080"
    restart: "no"
    env_file:
      - .env
    depends_on:
      postgres:
        condition: service_healthy
